### PR TITLE
LiveTV support on tvOS

### PR DESCRIPTION
**Pull request by jhays**
Adds support for navigating / displaying / playback of Live TV on tvOS. 
(I'll work on adding support to iOS next- should be relatively straightforward.)

LiveTV can be found under the main tab "Other" if the Jellyfin server has a library with the "livetv" content type. 

Selecting this will launch a new tab view with "Programs", "Channels", and "Home" tabs. Selecting Home returns to the main tab interface. 

Items from Programs and Channels can be selected to launch the video player playback interface. 

Feedback welcome on the code or design- 
If there are any preferences as far as on-brand colors or anything like that, I'm open to input. 

I'd prefer a default image instead of the all-white when an image cannot be found. At the moment I'm not sure why missing images end up all white. 

Future steps:

- iOS support for these views
- Consolidate channel / program fetching logic (there is currently some duplicated effort between both LiveTV related view models.)
- Add "Guide" EPG style view. I had initially started this, but it is fairly complex. I decided to get these views wrapped up first and defer the full EPG work until later. 
- Add ability to pull up a half-screen channel browser or convert the currently-player video to PiP to facilitate browsing channels while watching one. 

Simulator Screen Shot - Apple TV 4K (at 1080p) (2nd generation) - 2021-11-20 at 14 05 03

Simulator Screen Shot - Apple TV 4K (at 1080p) (2nd generation) - 2021-11-20 at 13 59 14

### CODE REVIEW

File: JellyfinPlayer tvOS/Views/LiveTVChannelsView.swift [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                EmptyView()
+                .accessibilityIdentifier("\(indexPath.section).\(indexPath.row)")
+            }
+            .frame(maxWidth: .infinity, maxHeight: .infinity)
+            .ignoresSafeArea()
+            
+        } else {
+            VStack {
+                Text("No results.")
+                Button {
```
Comment (LePips) [inline JellyfinPlayer tvOS/Views/LiveTVChannelsView.swift:38]: Is this button supposed to do something like a retry?

File: JellyfinPlayer tvOS/Views/LiveTVChannelsView.swift [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                    print("movieLibraries reload")
+                } label: {
+                    Text("Reload")
+                }
+            }
+        }
+    }
+    
+    @ViewBuilder func makeCellView(indexPath: IndexPath, cell: LiveTVChannelRowCell) -> some View {
+        GeometryReader { _ in
```
Comment (LePips) [inline JellyfinPlayer tvOS/Views/LiveTVChannelsView.swift:48]: Is there a reason to use a reader here without using the value of it?

**Review by LePips — CHANGES_REQUESTED** [review]

**Comment by maniacaris** [comment]
Thanks so much for your work on this. Live TV is currently one of my main use-cases for Jellyfin and the HDHomeRun + Infuse apps leave a lot to be desired. Please let me know if you need help testing. I have an HDHR Duo, Apple TV 4k, Gen 3 Apple TV, and Jellyfin running on Proxmox.

File: JellyfinPlayer tvOS/Views/LiveTVChannelsView.swift (hunk 1 above)
Comment (jhays) [reply JellyfinPlayer tvOS/Views/LiveTVChannelsView.swift:38]: Correct, that was the intention. Updated with latest commit to actually perform the retry.

File: JellyfinPlayer tvOS/Views/LiveTVChannelsView.swift (hunk 2 above)
Comment (jhays) [reply JellyfinPlayer tvOS/Views/LiveTVChannelsView.swift]: No- this wasn't actually needed. Removed in latest commit.

**Comment by jhays** [comment]
Just pushed the last commit I was planning for this branch- adds a timer that auto-refreshes the current program displayed for each channel on the channels view on every 10th minute. 

I plan to investigate adding liveTV to iOS next, but as a separate branch.

I don't have any other pending changes planned on this PR so I would consider this ready for final review & merge.

**Comment by AndrewBreyen** [comment]
Patiently waiting for this! this would be so amazing. We just got a HDHomeRun and AppleTV and this is one of our key use cases. Thanks!

**Comment by LePips** [comment]
I'll take a look at this soon

**Comment by LePips** [comment]
I was able to get a free m3u set up but wasn't able to play anything, even with streams that were working. I didn't look around too much but I think this is due to the VLCKit player currently set up. This overall looks good to me with just some visual things that could be fixed but I will get to those. 

I'd say that what is here is good for now and will merge and I will continue my player re-factor with live tv included. It shouldn't be too difficult as I've thought about it a lot.

**Review by LePips — APPROVED** [review]

**Comment by AndrewBreyen** [comment]
With a HDHOMERUN, I also wasn't able to get things to play, it just comes up "Empty" when something is selected. Happy to provide logs if needed!

**Comment by LePips** [comment]
Nothing is meant to be playing at the moment as a video player needs to be accommodated for live streams. He is readily working on this.

**Pull request merged**